src/main/java/com/smartcar/sdk/data/BatchResponse.java

```diff
@@ -0,0 +1,101 @@
+import com.google.gson.JsonParser;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.GsonBuilder;
+import org.apache.commons.text.CaseUtils;
+import java.util.HashMap;
+import java.util.Iterator;
+
+public class BatchResponse extends ApiData {
+  private HashMap<String, JsonObject> data = new HashMap<>();
+
+
+  public  BatchResponse(String responsesArray) {
+
+    JsonParser parser = new JsonParser();
+    JsonElement responsesParsed = parser.parse(responsesArray);
+    JsonArray responsesJsonArray = responsesParsed.getAsJsonArray();
+    Iterator<JsonElement> responses = responsesJsonArray.iterator();
+
+    while (responses.hasNext()){
+      JsonObject res = responses.next().getAsJsonObject();
+      String path = res.get("path").getAsString();
+      this.data.put(path, res);
+    }
+  }
+
+  private static String toCamelCase(String fieldName) {
+    if (fieldName.contains("_")) { // checks for snake case
+      return CaseUtils.toCamelCase(fieldName, false, new char[]{'_'});
+    }
+    return fieldName;
+  }
+
+  static GsonBuilder gson = new GsonBuilder().setFieldNamingStrategy((field) -> toCamelCase(field.getName()));
+
+
+  public <T extends ApiData> SmartcarResponse<T> get(String path) throws SmartcarException {
+    JsonObject res = this.data.get(path);
+    Integer code = res.get("code").getAsInt();
+    if (code != 200) {
+      // TODO : handle error
+      System.out.println("ERROR");
+    }
+    JsonObject body = res.get("body").getAsJsonObject();
+    JsonElement header = res.get("headers");
+    Class<T> dataType = getClassByPath(path);
+    return createSmartcarResponse(body, header, dataType);
+  }
+
+  private <T extends ApiData> SmartcarResponse<T> createSmartcarResponse(JsonElement body, JsonElement header, Class<T> dataType) throws SmartcarException {
+    T data = gson.create().fromJson(body, dataType);
+    System.out.println("fuck");
+    System.out.println(header);
+
+    String unitHeader = header.getAsJsonObject().get("sc-unit-system").getAsString();
+    String ageHeader;
+    try {
+      ageHeader = header.getAsJsonObject().get("sc-data-age").getAsString();
+    } catch(Exception e){
+      ageHeader = null;
+   }
+
+    if(ageHeader != null) {
+      DateTime date = DateTime.parse(ageHeader);
+      return new SmartcarResponse<T>(data, unitHeader, date.toDate());
+    } else {
+      return new SmartcarResponse<T>(data, unitHeader, null);
+    }
+  }
+
+  public <T extends ApiData> Class<T> getClassByPath(String path) throws SmartcarException {
+    switch (path) {
+      case "/odometer":
+        return (Class<T>) VehicleOdometer.class;
+      case "/battery":
+        return (Class<T>) VehicleBattery.class;
+      case "/charge":
+        return (Class<T>) VehicleCharge.class;
+      default:
+        return (Class<T>) VehicleFuel.class;
+      }
+  }
```

Comment:
@sankethkatta do you think there is a better way to do this than a straight-up switch statement? Should I throw if they ask for a path that is invalid?